api/types.proto

```diff
@@ -713,3 +727,31 @@ message ManagerStatus {
 	// Reachability specifies whether this node is reachable.
 	RaftMemberStatus.Reachability reachability = 4;
 }
+
+// SecretReference is the linkage between a service and a secret that it uses.
+message SecretReference {
+	// Name is the name of the secret that this references.
+	string name = 1;
+
+	// SecretValueID represents the ID of the specific Secret version that we're
+	// referencing. This identifier exists so that SecretReferences don't leak
+	// any information about the secret contents.
+	string secret_value_id = 2 [(gogoproto.customname) = "SecretValueID"];
```

Comment:
This is why we had an unlimited number of secrets, so we didn't have to deal with this particular issue ;)

Not sure if rejecting an update because it will make a service fail is the right thing to do. Technically people can also create services with the wrong image name, etc etc.

Maybe a warning should be printed if you're about to pop a secret that is in active use?